system/include/emscripten/val.h

```diff
@@ -265,9 +265,9 @@ EMSCRIPTEN_ALWAYS_INLINE void writeGenericWireTypes(GenericWireType*& cursor, Fi
 
 template<typename... Args>
```

Comment:
Just to make sure I understand this correctly... WirePack() has a fixed type (based on the struct) so std::foward is not needed here?

Reply:
As written, the `WireTypePack` constructor only takes (non-const) rvalue references, not the usual forwarding mix of either lvalue or rvalue references.  Those `args` then (in `writeGenericWireTypes`) get passed into `BindingType::toWreType`, and I thought that those `toWireType` functions are all happy with taking const lvalue references, anyway, so we could simplify the `WireTypePack` constructor to just take const lvalue references from the start.

But, on a second look, my reasoning was wrong, and at least the `toWireType` for `std::unique_ptr` requires an rvalue argument, so my proposed change here would actually break things.  (We could change the `std::forward` to `std::move`, to make it more obvious that no forwarding references are involved here, but the code is just fine as it is, too.)  So, sorry for the noise...